zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/client/UserClient.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.util.client;
+
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.UserIntent;
+import io.camunda.zeebe.protocol.record.value.UserRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import java.util.function.Function;
+
+public final class UserClient {
+
+  private final CommandWriter writer;
+
+  public UserClient(final CommandWriter writer) {
+    this.writer = writer;
+  }
+
+  public UserCreationClient newUser(final String username) {
+    return new UserCreationClient(writer, username);
+  }
+
+  public static class UserCreationClient {
+
+    private static final Function<Long, Record<UserRecordValue>> SUCCESS_SUPPLIER =
+        (position) -> RecordingExporter.userRecords().withSourceRecordPosition(position).getFirst();
```

Comment:
❌ You now return the first record of the ValueType user. This will always be the `CREATE` command. If you really want to expect a success you should wait for the `CREATED` event. If that record exists you know 100% that the user is created. Right now this expectation would also work for a command rejection.

It also means that in your current test cases you are testing the data you use as input. There's no verification of the date in the `CREATED` event. It's pretty simple to change this. Changing it to this should do the trick:

```java
RecordingExporter.userRecords().withIntent(UserIntent.CREATED).withSourceRecordPosition(position).getFirst();
```